packages/quantic/cypress/e2e/default-1/case-classification/case-classification.cypress.ts

```diff
@@ -61,66 +67,1498 @@ describe('quantic-case-classification', () => {
   function visitCaseClassification(
     options: Partial<CaseClassificationOptions>
   ) {
+    if (analyticsMode === 'next') {
+      setCookieToEnableNextAnalytics(exampleTrackingId);
+    }
     mockSfPicklistValues(sfDefaultField, allOptions);
     interceptCaseAssist();
     cy.visit(pageUrl);
     configure(options);
   }
 
-  describe('when using default options', () => {
-    it('should render the component and all parts', () => {
-      visitCaseClassification({});
+  analyticsModeTest.forEach((analytics) => {
+    describe(analytics.label, () => {
+      before(() => {
+        analyticsMode = analytics.mode;
+      });
+
+      describe('when using default options', () => {
+        it('should render the component and all parts', () => {
+          const exampleResponseId = crypto.randomUUID();
+          visitCaseClassification({});
+
+          scope('when loading the page', () => {
+            Expect.displayLabel(true);
+            Expect.numberOfInlineOptions(0);
+            Expect.numberOfSuggestions(0);
+            Expect.displaySelectTitle(false);
+            Expect.displaySelectInput(true);
+          });
+
+          scope('when fetching suggestions', () => {
+            const suggestionsCount = 3;
+            const firstSuggestionIndex = 0;
+
+            mockCaseClassification(
+              coveoDefaultField,
+              allOptions.slice(0, suggestionsCount),
+              exampleResponseId
+            );
+            fetchClassifications();
+            Expect.displaySelectTitle(true);
+            Expect.displaySelectInput(false);
+            Expect.numberOfSuggestions(suggestionsCount);
+            Expect.correctSuggestionsOrder(
+              allOptions.slice(0, suggestionsCount)
+            );
+            Expect.numberOfInlineOptions(0);
+            if (analyticsMode === 'next') {
+              NextAnalyticsExpectations.emitCaseAssistSelectFieldClassification(
+                {
+                  fieldClassification: {
+                    id: allOptions[firstSuggestionIndex].id,
+                    responseId: exampleResponseId,
+                  },
+                  autoselected: true,
+                },
+                exampleTrackingId
+              );
+            } else {
+              Expect.logClickedSuggestion(firstSuggestionIndex, true);
+            }
```

Comment:
Would it be better to organize things like this instead:

```
const analyticsExpectation = Expect.clickedSuggestionExpectation(analyticsMode);
if(analyticsMode === 'next') {
    analyticsExpectation({
        fieldClassification: {
            id: allOptions[firstSuggestionIndex].id,
            responseId: exampleResponseId,
        },
        autoselected: true,
    },
    exampleTrackingId);
} else {
    analyticsExpectation(firstSuggestionIndex, true);
}
```

And in the case-classification-expectations file, you could have the function `clickedSuggestionExpectation` return either the previous analytics expectations function, or the next-analytics-expectations function?

I think this way it would be easier later to remove the test in V3 for example if we want to stop testing for previous analytics?

It would be even better if we could be able to pass the same object to the `Expect.clickedSuggestionExpectations` so that the "body" of the expected analytics is built from the same object.
This way we could have perhaps

```
Expect.clickedSuggestionAnalyticsExpectations(dataToBuildTheExpectedBody, analyticsMode);
```